script/tool/lib/src/gradle_check_command.dart

```diff
@@ -120,6 +130,122 @@ class GradleCheckCommand extends PackageLoopingCommand {
     return succeeded;
   }
 
+  /// String printed as example of valid example root build.gradle repository
+  /// configuration that enables artifact hub env variable.
+  @visibleForTesting
+  static const String exampleRootGradleArtifactHubString = r'''
+        // See https://github.com/flutter/flutter/wiki/Plugins-and-Packages-repository-structure#gradle-structure for more info.
+        def artifactRepoKey = 'ARTIFACT_HUB_REPOSITORY'
+        if (System.getenv().containsKey(artifactRepoKey)) {
+            println "Using artifact hub"
+            maven { url System.getenv(artifactRepoKey) }
+        }
+''';
+
+  /// Validates that [gradleLines] reads and uses a artifiact hub repository
+  /// when ARTIFACT_HUB_REPOSITORY is set.
+  ///
+  /// Required in root gradle file.
+  bool _validateArtifactHubUsage(
+      RepositoryPackage example, List<String> gradleLines) {
+    const String keyVariable = 'artifactRepoKey';
+    final RegExp keyPresentRegex =
+        RegExp("$keyVariable\\s+=\\s+'ARTIFACT_HUB_REPOSITORY'");
+    final RegExp documentationPresentRegex = RegExp(
+        r'github.com.*wiki.*Plugins-and-Packages-repository-structure.*gradle-structure');
+    final RegExp keyReadRegex =
+        RegExp('if.*System.getenv.*containsKey.*$keyVariable');
+    final RegExp keyUsedRegex =
+        RegExp('maven.*url.*System.getenv.*$keyVariable.*');
+
+    final bool keyPresent =
+        gradleLines.any((String line) => keyPresentRegex.hasMatch(line));
+    final bool documentationPresent = gradleLines
+        .any((String line) => documentationPresentRegex.hasMatch(line));
+    final bool keyRead =
+        gradleLines.any((String line) => keyReadRegex.hasMatch(line));
+    final bool keyUsed =
+        gradleLines.any((String line) => keyUsedRegex.hasMatch(line));
+
+    if (!documentationPresent) {
+      printError('Does not link artifact hub documentation.');
```

Comment:
After seeing this output run on packages missing artifact hub implementation I removed documentation as a separate printed error since it is included in the combined string.